js_modules/dagit/packages/core/src/app/PythonErrorInfo.tsx

```diff
@@ -7,20 +7,27 @@ import {MetadataEntries} from '../metadata/MetadataEntry';
 import {MetadataEntryFragment} from '../metadata/types/MetadataEntryFragment';
 import {ErrorSource} from '../types/globalTypes';
 
-import {PythonErrorFragment} from './types/PythonErrorFragment';
+import {
+  PythonErrorFragment,
+  PythonErrorFragment_causes as Cause,
+} from './types/PythonErrorFragment';
+
+export type GenericError = {
+  message: string;
+  stack?: string[];
+  causes?: Cause[];
+};
 
 interface IPythonErrorInfoProps {
   showReload?: boolean;
   centered?: boolean;
-  error: {message: string} | PythonErrorFragment;
+  error: GenericError | PythonErrorFragment;
   failureMetadata?: {metadataEntries: MetadataEntryFragment[]} | null;
   errorSource?: ErrorSource | null;
 }
 
 export const PythonErrorInfo: React.FC<IPythonErrorInfoProps> = (props) => {
-  const message = props.error.message;
-  const stack = (props.error as PythonErrorFragment).stack;
-  const causes = (props.error as PythonErrorFragment).causes;
+  const {message, stack = [], causes = []} = props.error;
```

Comment:
this is ultimately the only non-typing change right? None will be cast to empty list now

Reply:
No casting anywhere now. `causes` can now be an array or `undefined` (because of the `GenericError | PythonErrorFragment` union, so in this case we're just setting the undefined value (a `GenericError`) to an empty array so that `map` is always called on an array.